ui/raidboss/timeline.js

```diff
@@ -202,23 +201,28 @@ export class Timeline {
       let match = regexes.ignore.exec(line);
       if (match && match['groups']) {
         const ignore = match['groups'];
-        this.ignores[ignore.id] = true;
-        continue;
+        if (ignore.id) {
+          this.ignores[ignore.id] = true;
+          continue;
+        }
       }
 
       match = regexes.tts.exec(line);
       if (match && match['groups']) {
         const tts = match['groups'];
-        // TODO: Support alert sounds?
-        if (tts.command === 'sound')
+        if (tts.id && tts.beforeSeconds && tts.command) {
```

Comment:
```suggestion
        // Convince TypeScript these exist.
        if (tts.id && tts.beforeSeconds && tts.command)
          continue;
```

(+ early return instead of more nesting, here and elsewhere)

Reply:
(or feel free not to leave a comment? Maybe it's weird to drop these comments everywhere, but I think there's something useful to me in knowing "me as the programmer really expects this path to not be possible given what I know" vs "this is a check I expect to possibly fire in practice"  I don't know how to make this more readable?  Do we need a `convinceTypeScript() { throw new Error('This shouldn\'t happen OOPS'); }`?  Other ideas???)

Reply:
I don't feel like "Convince TypeScript these exist" is the right call here, though. Right? There's no explicit contract that says these values have to exist. I think it would be more appropriate to log an error and continue. How do you feel about that?

Reply:
There is no explicit contract, but I think logically this regex has no optional capture groups.  If it matches, then all of these fields are defined.  Arguably, somebody could change the regex and not this code, and maybe that's what you're getting at?

What about rephrasing it as `notReached() { throw new Error('This code shouldn\'t be reached'); }`? Or do you think an exception is too heavy?

Maybe I'm overthinking this?

Reply:
Added!